plugins/content/tjreportsfields/tjreportsfields.php

```diff
@@ -0,0 +1,289 @@
+	 * This event is fired before the data is actually saved.
+	 *
+	 * @param   string   $context  The context of the content passed to the plugin.
+	 * @param   JTable   $row      A JTable object.
+	 * @param   boolean  $isNew    If the content is just about to be created.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   2.5
+	 * @throws  Exception on database error.
+	 */
+	public function onContentBeforeSave($context, $row, $isNew)
+	{
+		// $context = com_fields.field
+
+		JTable::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_fields/tables');
+		$fieldsTable = JTable::getInstance('Field', 'FieldsTable');
+		$fieldsTable->load(array('id' => $row->id));
+
+		// Set an array with field id, field name, so it can be used in onContentAfterSave trigger
+		// Array ( [3] => dob )
+		$this->customFieldBeingEdited       = new stdclass;
+		$this->customFieldBeingEdited->id   = $row->id;
+		$this->customFieldBeingEdited->name = $fieldsTable->name;
+		$this->customFieldBeingEdited->type = $fieldsTable->type;
+
+		// @echo '<br/>' . $context; echo '<br/>';  var_dump($row->id); print_r($this->customFieldBeingEdited); die;
+
+		return true;
+	}
+
+	/**
+	 * Smart Search after save content method.
+	 * Content is passed by reference, but after the save, so no changes will be saved.
+	 * Method is called right after the content is saved.
+	 *
+	 * @param   string  $context  The context of the content passed to the plugin (added in 1.6)
+	 * @param   object  $field    A JTableField object
+	 * @param   bool    $isNew    If the content has just been created
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0.0
+	 */
+	public function onContentAfterSave($context, $field, $isNew)
+	{
+		// Run this plugin only for com_fields.field form
+		if ($context !== 'com_fields.field')
+		{
+			return;
+		}
+
+		/*
+		[id] => 2
+		[context] => com_users.user
+		[group_id] => 1
+		[title] => Professional Details - Job Start Year
+		[name] => job-start-year
+		[label] => Professional Details - Job Start Year
+		[default_value] =>*/
+
+		// Skip unsupported field
+		if (in_array($field->type, $this->unSupportedFields))
+		{
+			return;
+		}
+
+		// Set table name as #__tjreports_context eg: #__tjreports_com_users_user
+		$this->customFieldsTable = '#__tjreports_' . str_replace('.', '_', trim($field->context));
+
+		// If no table, return
+		if (!$this->tableExists())
+		{
+			return;
+		}
+
+		// Get column name, type for custom fields index table
+		$db             = JFactory::getDbo();
+		$columnsDetails = $db->getTableColumns($this->customFieldsTable);
+
+		// Extract column names from $columnDetails
+		$columnNames = array();
+
+		foreach ($columnsDetails as $key => $val)
+		{
+			$columnNames[] = $key;
+		}
+
+		$oldColumnName = $this->customFieldBeingEdited->name;
+		$newColumnName = $field->name;
+
+		// If current being edited already exits in indexed DB
+		if (in_array($oldColumnName, $columnNames))
+		{
+			// If current filed name is same, do nothing
```

Comment:
Remove this `if` since there's no code inside.

Reply:
oops , yh